modules/database/funcs.go

```diff
@@ -229,3 +231,34 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+//
+// Returns a *Database object with specified name and namespace
+//
+func GetDatabaseByName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+) *Database {
+	// you can create a Database here by only filling what we need (name and namespace)
+	db := &Database{
+		databaseName: name,
+	}
+	// then querying the MariaDBDatabase and store it in db by calling
+	db.getDBWithName(ctx, h)
```

Comment:
you need to handle the possible error returned here

Reply:
ok , since the type of error is handled in getDBWithName I think returning an err or nil logic is ok here